add xgboost server rock and tests

ca-scribner opened the pull request:
Closes #26.

### Testing instructions:
See xgboost/readme.md

orfeas-k commented on xgbserver/rockcraft.yaml:
```diff
@@ -0,0 +1,78 @@
+    source-subdir: python
+    source-tag: v0.11.2
+    build-packages:
+      - libgomp1
```
I see there is no `build-essential` package compared to other similar ROCKs, although it is being used in the upstream [dockerfile](https://github.com/kserve/kserve/blob/release-0.11/python/xgb.Dockerfile#L12). My guess is that it's okay without it too and I don't see it as a blocker, since the ROCK works as expected. But let's verify that this is intentional.

orfeas-k commented on xgbserver/rockcraft.yaml:
```diff
@@ -0,0 +1,78 @@
+      # Install the kserve package, this specific server package, and their dependencies.
+      mkdir -p ./python_env_builddir
+      cp -rf $CRAFT_PROJECT_DIR/dummy_pyproject.toml ./python_env_builddir/pyproject.toml
+      (cd python_env_builddir && poetry install --no-interaction --no-root)
```
Why is this inside parentheses? I 'm [reading](https://ss64.com/bash/syntax-brackets.html):

> Placing a list of commands between parentheses causes a [subshell](https://unix.stackexchange.com/a/138498) to be created, and each of the commands in list to be executed in that subshell, without removing non-[exported](https://ss64.com/bash/export.html) variables.
Since the list is executed in a subshell, variable assignments do not remain in effect after the subshell completes.

But I don't see the benefit since the first command is just a `cd` command.

ca-scribner replied:
We can remove them.  I didn't add this myself, but I believe the intent is so we don't need to `cd` back to the starting dir after.  For example:

```
mkdir -p /tmp/start
mkdir -p /tmp/another_location
cd /tmp/start

(cd /tmp/another_location && echo "in $(pwd)")
# in /tmp/another_location
echo "in $(pwd)"
# in /tmp

cd /tmp/another_location && echo "in $(pwd)"
# in /tmp/another_location
echo "in $(pwd)"
# in /tmp/another_location
```

We can switch it to another method if that's better though.  Could be that we `cd ..` after.  This is just a way where we don't need to care about it as much

orfeas-k replied:
oh good point actually. Let's keep it as is

orfeas-k reviewed (commented):
Everything looks good. I left two small comments. 

Btw, the only difference in response I noticed when testing is 

```diff
-  "model_version": "v1.0.0",
+  "model_version": null,
+  "parameters": null,
```
We 've seen this happen in other places where we use ROCKs like https://github.com/canonical/seldonio-rocks/issues/71 and I 'm still wondering why, but yeah it's not a blocker. I just thought of mentioning it.

orfeas-k commented on xgbserver/dummy_pyproject.toml:
```diff
@@ -0,0 +1,11 @@
+authors = ["none"]
+
+[tool.poetry.dependencies]
+python = ">=3.8,<3.12"
```
Should we have a [comment like in other ROCKs](https://github.com/canonical/kserve-rocks/blob/main/pmmlserver/dummy_pyproject.toml#L8)?

orfeas-k commented:
Also, it would be nice if we had the `testing` instructions automated as sanity tests.

ca-scribner commented:
@orfeas-k agreed about the testing instructions.  I actually looked into it, but didn't find a simple way so I dropped it for now.  The thing I was hung up on was trying to do it from tox, and I wasn't sure how to  `docker run` the thing and then curl at it.  But it would help a lot if they were automated - I just added #47 to capture this

orfeas-k reviewed (approved):
thank you @ca-scribner
